app/controllers/users_controller.rb

```diff
@@ -12,18 +16,72 @@ def new
   def create
     @user = User.new user_params
     if @user.save
-      flash[:info] = t ".create_user_message"
+      save_avatar
+      flash[:success] = t ".create_user_message"
       redirect_to root_url
     else
       flash.now[:danger] = t ".create_user_error"
       render :new
     end
   end
 
+  def edit; end
+
+  def update
+    if @user.update(user_params)
+      flash[:success] = t ".edit_success_message"
+      redirect_to @user
+    else
+      flash.now[:danger] = t ".edit_failure_message"
+      render :edit
+    end
+  end
+
+  def destroy
+    if @user.destroy
+      flash[:success] = t ".deleted_message"
+    else
+      flash[:danger] = t ".delete_failed_message"
+    end
+    redirect_to users_path
+  end
+
   private
 
   def user_params
     params.require(:user).permit(:full_name, :email, :password,
-                                 :password_confirmation)
+                                 :password_confirmation, :avatar)
+  end
+
+  def find_user
+    @user = User.find_by id: params[:id]
+    return if @user
+
+    flash[:danger] = t ".not_found"
+    redirect_to root_path
+  end
+
+  def paginate_users
+    @pagy, @users = pagy filter_user, items: Settings.user.per_page
+  end
+
+  def filter_user
+    return User.sort_created_at unless params[:filter]
+
+    @users = User.by_email(params[:filter][:email_search])
+                 .by_full_name(params[:filter][:full_name_search])
+                 .sort_created_at
+  end
+
+  def correct_user
+    redirect_to(root_path) unless current_user?(@user)
+  end
+
+  def admin_user
+    redirect_to(root_path) unless current_user.admin?
+  end
+
+  def save_avatar
+    @user.avatar.attach(params[:user][:avatar]) if params[:user][:avatar]
```

Comment:
có 1 dòng tách hàm làm gì e?

Reply:
Vì trong trường hợp không tách hàm thì rubocop cảnh báo: **"Assignment Branch Condition size for create is too high".** Vậy nên em đã tiến hành tách hàm ra ạ.